src/data/real/sqrt.lean

```diff
@@ -272,6 +272,27 @@ by rw [← not_le, not_iff_not, sqrt_eq_zero']
 lt_iff_lt_of_le_iff_le (iff.trans
   (by simp [le_antisymm_iff, sqrt_nonneg]) sqrt_eq_zero')
 
+alias sqrt_pos ↔ _ sqrt_pos_of_pos
+
+section
+open tactic tactic.positivity
+
+/-- Extension for the `positivity` tactic: a square root is nonnegative, and is strictly positive if
+its input is. -/
+@[positivity]
+meta def _root_.tactic.positivity_sqrt : expr → tactic strictness
+| `(real.sqrt %%a) := do
+  (do -- if can prove `0 < a`, report positivity
+    strictness_a ← core a,
+    match strictness_a with
+    | (positive pa) := positive <$> mk_app ``sqrt_pos_of_pos [pa]
+    | (nonnegative _) := failed
+    end) <|>
```

Comment:
I think this golfs to
```suggestion
  (do -- if can prove `0 < a`, report positivity
    positive pa ← core a,
    positive <$> mk_app ``sqrt_pos_of_pos [pa]) <|>
```

Reply:
It does, but in my limited tactic knowledge I had the idea that it was less expensive to fail after an unsuccessful match than to fail after an unsuccessful `mk_app`.  Is that not the case?

Reply:
I don't understand the distinction you're making; I think the only difference between yours and mine is that mine gives the error "match failed" when the match fails, rather than "failed".

Reply:
I had the idea that, in the nonnegative case, it cost a little bit of time to try to apply the lemma `sqrt_pos_of_pos` to the proof `pa`, and have Lean discover that actually the lemma was not applicable -- and that it would be faster for Lean to simply fail directly in that case by sending it there in a match.  No?

Reply:
In the nonnegative case the pattern match fails in my version in the same way as it fails for yours; `positive pa ← core a, foo` is syntax for
```lean
strictness_a ← core a,
match strictness_a with
| (positive pa) := foo
| (nonnegative _) := fail!"match failed"
end
```

Reply:
Ah, I see, I misunderstood your syntax.  Thanks, I'll change!  The same applies to the `abs` tactic and I'll also change there; let me know if you spot other places I should do this.